### PR TITLE
Gp19 60 create patch endpoint

### PR DESCRIPTION
**Pull request by sophiambolah**

### CODE REVIEW

File: SpringProject/src/it/java/com/gradproject2019/ConferenceControllerIT.java [hunk 1]
```diff
@@ -161,36 +169,78 @@ public void shouldReturn404WhenConferenceToBeDeletedNotFound() throws URISyntaxE
...
+        Assert.assertEquals(404, response.getStatusCodeValue());
     }
 
+    @Test   //WHEN CUSTOM HIBERNATE QUERY WRITTEN, THIS TEST WILL PASS
+    public void shouldReturn200AndEditOnlyNotNullFields() throws URISyntaxException {
+        //given
+        Conference savedConference = conferenceRepository.saveAndFlush(conference);
+        URI uri = new URI(baseUrl + "/" + savedConference.getId());
+        String newName= "Harry's conference";
+        String newCity= "Manchester/North";
```
Comment (H-Ford) [inline SpringProject/src/it/java/com/gradproject2019/ConferenceControllerIT.java:203]: 😂

**Review by Piotrek605 — APPROVED** [review]

File: SpringProject/src/main/java/com/gradproject2019/conferences/service/ConferenceServiceImpl.java [hunk 2]
```diff
@@ -60,4 +63,13 @@ private void checkConferenceExists(Long conferenceId) {
             throw new ConferenceNotFoundException();
         }
     }
+
+    @Override
+    public ConferenceResponseDto editConference(Long conferenceId, ConferencePatchRequestDto conferencePatchRequestDto){
```
Comment (H-Ford) [inline SpringProject/src/main/java/com/gradproject2019/conferences/service/ConferenceServiceImpl.java]: This method needs to be above the private methods, the order of methods should be public ones first followed by your private methods

File: SpringProject/src/main/java/com/gradproject2019/conferences/service/ConferenceServiceImpl.java [hunk 3]
```diff
@@ -44,9 +45,11 @@ public ConferenceResponseDto saveConference(ConferenceRequestDto conferenceReque
     }
 
     private void checkNotInPast(Conference conference) {
```
Comment (H-Ford) [inline SpringProject/src/main/java/com/gradproject2019/conferences/service/ConferenceServiceImpl.java]: You should just pass the conferences dateTime here, no need to pass the whole conference

File: SpringProject/src/main/java/com/gradproject2019/conferences/service/ConferenceServiceImpl.java [hunk 4]
```diff
@@ -60,4 +63,13 @@ private void checkConferenceExists(Long conferenceId) {
             throw new ConferenceNotFoundException();
         }
     }
+
+    @Override
+    public ConferenceResponseDto editConference(Long conferenceId, ConferencePatchRequestDto conferencePatchRequestDto){
+        checkConferenceExists(conferenceId);
+        Conference conference = conferencePatchRequestDto.from(conferenceId, conferencePatchRequestDto);
```
Comment (H-Ford) [inline SpringProject/src/main/java/com/gradproject2019/conferences/service/ConferenceServiceImpl.java]: You don't need to convert the conferencePatchRequestDto to a Conference here, you can do everything you need without doing that

**Review by H-Ford — COMMENTED** [review]
Looks pretty good, especially the tests 😄 Remeber to remove unused imports, and don't leave unnecessary blank lines!

**Review by singhals-els — APPROVED** [review]

**Review by H-Ford — APPROVED** [review]

**Pull request merged**